### PR TITLE
[WIP] Migration Document for ASP.Net Framework to ASP.Net Core 2.0

### PR DESCRIPTION
**Pull request by isaac2004**
[Internal Review URL](https://review.docs.microsoft.com/aspnet/core/migration/proper-to-2x/mvc2?branch=pr-en-us-3897)

Document created for #3809, and currently being completed, please review

@scottaddie @Rick-Anderson

### CODE REVIEW

File: aspnetcore/migration/mvc2.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+## Static file / wwwroot
+One essential part of web development is the ability to serve files that are "static" or in better terms, not built code to the browser. The most common examples of static files are HTML, CSS, Javascript and images and they are used by every website on the internet. We need to be able to save these files in the published location of our application (or CDN) and reference these files to load onto the browser. This process has changed in Dot Net Core wheras in previous versions of the .Net Framework, you could store your static files basically anywhere, and reference them in your view like regular html (with the path to the asset provided). 
+
+In Dot Net Core, we store our static files in `web root` (<content root>/wwwroot) and loading them into the pipeline using the `UseStaticFiles` extension from `Statup.Configure`. Be sure to include the NuGet package "Microsoft.AspNetCore.StaticFiles" in your application.
```
Comment (scottaddie) [inline aspnetcore/migration/mvc2.md]: - Dot Net Core --> .NET Core
- remove the back ticks surrounding "web root"
- "Microsoft.AspNetCore.StaticFiles" --> `Microsoft.AspNetCore.StaticFiles`

File: aspnetcore/migration/mvc2.md [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+[!code-csharp[Main](../fundamentals/static-files/sample/StartupStaticFiles.cs?highlight=3&name=snippet1)]
+
+Doing this allows all your assets in the wwwroot folder accessible to the browser at a location such as
```
Comment (scottaddie) [inline aspnetcore/migration/mvc2.md]: Italicize "wwwroot"

File: aspnetcore/migration/mvc2.md [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+`http://<app>/images/<imageFileName>` (This example has an image folder in wwwroot and that folder has images in it)
```
Comment (scottaddie) [inline aspnetcore/migration/mvc2.md]: Italicize "wwwroot"

**Comment by isaac2004** [comment]
@scottaddie please review

**Comment by scottaddie** [comment]
@isaac2004 Reviewing right now. Thank you for letting me know.

**Comment by scottaddie** [comment]
@Rick-Anderson Can you please review? Once this doc is good, I'll handle the associated TOC updates separately.

**Comment by ardalis** [comment]
Do we include adjusting DbContext constructors anywhere? See: https://twitter.com/cwoodruff/status/901101305657208832

**Comment by Rick-Anderson** [comment]
@isaac2004  Fantastic work! I did a copy edit - please check my changes.

**Comment by scottaddie** [comment]
@isaac2004 I'm requesting one more change before approving and merging this PR. Your *Startup.cs* file sample in the **Storing Configurations** section should look like the following:

```csharp
        public Startup(IConfiguration configuration)
        {
            Configuration = configuration;
        }

        public IConfiguration Configuration { get; }
```

With the 2.0 release, the boilerplate configuration code is done for you behind the scenes.

**Comment by isaac2004** [comment]
@Rick-Anderson changes look good, thanks
@scottaddie requested changes made
@ardalis That is an interesting point, I can add it in here pretty quickly if @scottaddie and @Rick-Anderson agree

**Pull request merged**